artifactory/utils/golang/project/project.go

```diff
@@ -102,10 +118,57 @@ func (project *goProject) PublishPackage(targetRepo, buildName, buildNumber stri
 	if err != nil {
 		return err
 	}
+	// Create the info file if Artifactory version is 6.10.0 and above.
+	version := version.NewVersion(_go.ArtifactoryMinSupportedVersionForInfoFile)
+	artifactoryVersion, err := servicesManager.GetConfig().GetArtDetails().GetVersion()
+	if err != nil {
+		return err
+	}
+	if version.AtLeast(artifactoryVersion) {
+		pathToInfo, err := project.createInfoFile()
+		if err != nil {
+			return err
+		}
+		defer os.Remove(pathToInfo)
+		if len(buildName) > 0 && len(buildNumber) > 0 {
+			err = project.addInfoFileToBuildInfo(pathToInfo)
+			if err != nil {
+				return err
+			}
+		}
+		params.InfoPath = pathToInfo
+	}
 
 	return servicesManager.PublishGoProject(params)
 }
 
+// Creates the info file.
+// Returns the path to that file.
+func (project *goProject) createInfoFile() (string, error) {
```

Comment:
This function has a few CheckError calls missing.

Reply:
Done